helpers/jwt-service.go

```diff
@@ -0,0 +1,40 @@
+package helpers
+
+import (
+	"os"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/joho/godotenv"
+)
+
+type JWTService interface {
+	GenerateToken(user_id string) (string, error)
+}
+
+type JWTServiceInfo struct {
+	secretKey string
+	issuer    string
+}
+
+func NewJWTService() JWTService {
```

Comment:
Why do we need to go through the effort of making a interface/struct with the `NewJWTService` function? Seems like this pattern is used for controllers, but unnecessary for this case? `secretKey` can be read inside the `GenerateToken` function (doesn't have to be instance data on `jwtSrv`. Can we just export `GenerateToken` as a pure function (you can inline the secret key). 

Doing this would also eliminate this change you made:
```diff
- func NewAuthController() AuthController {
-   return &authController{}
+ func NewAuthController(jwtService helpers.JWTService) AuthController {
+  return &authController{jwtService: jwtService}
+ }
```

Reply:
Mostly so that we don't have to unnecessarily load the secret key from the .env file **every** time the Login route is invoked. But I agree, this might be overkill, so I'll retrieve the secret key in NewAuthController() instead and pass that to the service, instead of NewJWTService.

Reply:
@anjanbharadwaj if loading from the .env file is expensive, store it as a constant in the root scope of jwt-service (`const SECRET = os.env('JWT_SECRET'`). It definitely seems more weird to pass in the secret to a controller. I don't think anything should need to be passed to the controller at all.